### PR TITLE
Add flake8-simplify to CI

### PR DESCRIPTION
**Pull request by berland**
**Issue**
Resolves all remaning flake8-simplify issues.

**Approach**
Do as suggested by flake8, or dodge it.

Recommended to review commit by commit.

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Updated documentation
- [ ] Ensured new behaviour is tested

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: src/ert/runpaths.py [hunk 1]
```diff
@@ -74,7 +74,8 @@ def write_runpath_list(
         :param iteration_numbers: The list of iterations to write entries for
         :param realization_numbers: The list of realizations to write entries for
         """
-        with self._create_and_open_file() as f:
+        Path(self.runpath_list_filename).parent.mkdir(parents=True, exist_ok=True)
```
Comment (xjules) [inline src/ert/runpaths.py:77]: This was suggested by the `flake8-simplify` ? interesting ...

File: src/ert/shared/share/ert/forward-models/res/script/rms.py [hunk 2]
```diff
@@ -131,7 +131,7 @@ def init_seed(self, iens):
                 random.seed()
                 seed = random.randint(0, RMSRun._max_seed)
 
-        self.seed = seed % RMSRun._max_seed
+        self.seed = seed % RMSRun._max_seed  # noqa
```
Comment (xjules) [inline src/ert/shared/share/ert/forward-models/res/script/rms.py:134]: So I guess this formatting (% value) is deprecated. What about just taking all those input strings and run:
`new_format_string = input_string.replace('%d', '{}')`
and just use `format` instead?

File: tests/unit_tests/config/test_observations.py [hunk 3]
```diff
@@ -142,11 +142,10 @@ def test_observations(minimum_case):
         values.append((index, value, std))
 
     test_vector = observations[observation_key]
-    index = 1
-    for node in test_vector:
+
+    for index, node in enumerate(test_vector):
```
Comment (xjules) [inline tests/unit_tests/config/test_observations.py:146]: :smile: this really finds nice staff!

File: tests/unit_tests/gui/test_gui_load.py [hunk 4]
```diff
@@ -208,11 +208,12 @@ def test_that_the_ui_show_warnings_when_there_are_no_observations(qapp, tmp_path
...
-        with open("poly-no-gen-kw.ert", "w", encoding="utf-8") as fout:
-            for line in fin:
-                if "GEN_KW" not in line:
-                    fout.write(line)
+    with open("poly.ert", "r", encoding="utf-8") as fin, open(
+        "poly-no-gen-kw.ert", "w", encoding="utf-8"
+    ) as fout:
+        for line in fin:
+            if "GEN_KW" not in line:
+                fout.write(line)
```
Comment (xjules) [inline tests/unit_tests/gui/test_gui_load.py:216]:
```python
`filtered_lines = [line for line in lines if "GEN_KW" not in line]`
fout.writelines(filtered_lines)
``` ?

File: src/ert/shared/share/ert/forward-models/res/script/rms.py (hunk 2 above)
Comment (berland) [reply src/ert/shared/share/ert/forward-models/res/script/rms.py:134]: I thought about such hacks, but I think the real issue is that we are using `%d`. As long as we use that, I think it is fair to demonstrate that by use of the modulo operator.

**Pull request merged**